game/kernel/common/kmachine.cpp

```diff
@@ -348,25 +350,91 @@ void DecodeTime(u32 ptr) {
   sceCdReadClock(clock.c());
 }
 
+void vif_interrupt_callback(int bucket_id) {
+  // added for the PC port for faking VIF interrupts from the graphics system.
+  if (vif1_interrupt_handler && MasterExit == RuntimeExitStatus::RUNNING) {
+    call_goal(Ptr<Function>(vif1_interrupt_handler), bucket_id, 0, 0, s7.offset, g_ee_main_mem);
+  }
+}
+
 /*!
- * PC PORT FUNCTIONS BEGIN
- */
-/*!
- * Get a 300MHz timer value.
- * Called from EE thread
+ * Added in PC port.
  */
+u32 offset_of_s7() {
+  return s7.offset;
+}
+
+/// PC PORT FUNCTIONS BEGIN
+
+/// <summary>
+/// Get a 300MHz timer value. Called from EE thread
+/// </summary>
 u64 read_ee_timer() {
   u64 ns = ee_clock_timer.getNs();
   return (ns * 3) / 10;
 }
 
-/*!
- * Do a fast memory copy.
- */
+/// <summary>
+/// Do a fast memory copy.
+/// </summary>
+/// <param name="dst"></param>
+/// <param name="src"></param>
+/// <param name="size"></param>
```

Comment:
I'd rather stick with the Doxygen-style (`/*! ...`) comments if possible. I've only seen this `<summary> </summary>` style used in C#.